openmdao/core/tests/test_check_derivs.py

```diff
@@ -2237,6 +2237,40 @@ def test_set_method_and_step_bug(self):
 
 class TestProblemCheckTotals(unittest.TestCase):
 
+    def test_check_totals_linear_constraint(self):
+        prob = om.Problem()
+        prob.model = SellarDerivatives()
+        prob.model.nonlinear_solver = om.NonlinearBlockGS()
+
+        prob.model.add_design_var('x', lower=-100, upper=100)
+        prob.model.add_design_var('z', lower=-100, upper=100)
+        prob.model.add_objective('obj')
+        prob.model.add_constraint('con1', upper=0.0)
```

Comment:
should we be testing a linear constraint here?  Perhaps change the upper bound of z to be a linear constraint?